crates/router/src/connector/nexinets/transformers.rs

```diff
@@ -54,61 +130,236 @@ pub struct NexinetsAuthType {
+                resource_id: types::ResponseId::ConnectorTransactionId(item.response.order_id),
+                redirection_data,
+                mandate_reference: None,
+                connector_metadata: Some(connector_metadata),
+            }),
+            ..item.data
+        })
+    }
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexinetsCaptureOrVoidRequest {
+    pub initial_amount: i64,
+    pub currency: String,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexinetsOrder {
+    pub order_id: String,
+}
+
+impl TryFrom<&types::PaymentsCaptureRouterData> for NexinetsCaptureOrVoidRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsCaptureRouterData) -> Result<Self, Self::Error> {
+        Ok(Self {
+            initial_amount: item.request.amount_to_capture,
+            currency: item.request.currency.to_string(),
+        })
+    }
+}
+
+impl TryFrom<&types::PaymentsCancelRouterData> for NexinetsCaptureOrVoidRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsCancelRouterData) -> Result<Self, Self::Error> {
+        Ok(Self {
+            initial_amount: item
+                .request
+                .amount
+                .ok_or(errors::ConnectorError::RequestEncodingFailed)?,
+            currency: item
+                .request
+                .currency
+                .ok_or(errors::ConnectorError::RequestEncodingFailed)?
+                .to_string(),
+        })
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexinetsPaymentResponse {
+    pub transaction_id: Option<String>,
+    pub status: NexinetsPaymentStatus,
+    pub order: NexinetsOrder,
+    #[serde(rename = "type")]
+    pub transaction_type: NexinetsTransactionType,
+}
+
+impl<F, T>
+    TryFrom<types::ResponseRouterData<F, NexinetsPaymentResponse, T, types::PaymentsResponseData>>
+    for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<F, NexinetsPaymentResponse, T, types::PaymentsResponseData>,
+    ) -> Result<Self, Self::Error> {
+        let transaction_id = item.response.transaction_id;
+        let connector_metadata = serde_json::to_value(NexinetsPaymentsMetadata { transaction_id })
+            .into_report()
+            .change_context(errors::ConnectorError::ResponseHandlingFailed)?;
+        Ok(Self {
+            status: enums::AttemptStatus::foreign_from((
+                item.response.status,
+                item.response.transaction_type,
+            )),
+            response: Ok(types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::ConnectorTransactionId(
+                    item.response.order.order_id,
```

Comment:
why are we using order_id instead of transaction_id?

Reply:
We need both order_id and transaction_id for subsequent actions in payment_attempt